### PR TITLE
[NPUW] Fixed chunked prefill with new IR version of LFM2

### PR DESCRIPTION
**Pull request by AsyaPronina**
### Details:
 - *Fixed chunked prefill with new IR version of LFM2*:
   - *New IR version wires `attention_mask` with Conv states by slicing that `attention_mask` from the left on the size of current prompt.*
   -  *NPUW places new tokens in the right end of the `input_ids` and `attention_mask` in static shape scenario, so mask should be sliced from the right end instead.*

### Tickets:
 - *CVS-190129*
 - *CVS-190528*

### AI Assistance:
 - *AI assistance used: yes*
 - *Assistant is used for parallel debugging, small issues correction in the fix and tests*

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 1]
```diff
@@ -813,6 +816,8 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
     } else {
         LOG_DEBUG("Adding position_ids input in case it doesn't exist in model: LFM-2 case.");
         ov::npuw::AddPositionIdsParam().run_on_model(kvcache_model);
+        LOG_DEBUG("Right-align attention_mask slice for Conv operations: LFM-2 case.");
+        ov::npuw::RightAlignMaskSliceForConv().run_on_model(kvcache_model);
```
Comment (eshiryae) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:818]: Overall looks good, but as this pass applied in the common pipeline (without any conditions) so i'd suggest to check that it's not affecting any other model (at least waiting for Validate run to finish)

Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:818]: Thanks a lot, Ekaterina! Waiting Validation's run!

File: src/plugins/intel_npu/src/plugin/npuw/npuw_transformations/right_align_mask_slice_for_conv.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        auto transpose = opp::wrap_type<ov::op::v1::Transpose>({matmul, opp::any_input()});
+        // VariadicSplit into B,C,~h in LFM2.
+        auto variadic_split =
+            opp::wrap_type<ov::op::v1::VariadicSplit>({transpose, opp::any_input(), opp::any_input()});
+
+        auto callback = [=](opp::Matcher& m) {
+            auto& node_to_output = m.get_pattern_value_map();
+
+            auto matched_attention_mask = node_to_output.at(attention_mask).get_node_shared_ptr();
+            const auto any_name = matched_attention_mask->output(0).get_any_name();
```
Comment (AlexanderKalistratov) [inline src/plugins/intel_npu/src/plugin/npuw/npuw_transformations/right_align_mask_slice_for_conv.cpp]: We should avoid using `get_any_name`. I think it is kinda depricated.

Please use `get_names` and search in it.
It happens in compilation time and is not performance critical

**Review by AlexanderKalistratov — APPROVED** [review]

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp (hunk 1 above)
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:818]: Validation shows no regressions!

File: src/plugins/intel_npu/src/plugin/npuw/npuw_transformations/right_align_mask_slice_for_conv.cpp (hunk 2 above)
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/npuw_transformations/right_align_mask_slice_for_conv.cpp]: Fixed, thanks a lot!
